data-platform/data-science-and-ml/oracle-data-science-service/code-assets/ods-mlflow.auth.mysql.demo.ipynb

```diff
@@ -0,0 +1,362 @@
+   "id": "4a0dbdbc",
+   "metadata": {},
+   "source": [
+    "![banner.png](attachment:c8f5c446-1276-449a-a395-418e60ad106d.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "66a06459",
+   "metadata": {},
+   "source": [
+    "</br><h2 style='color:#005EB8'>ML Flow - Using OCI Tracking Server with Auth Enabled and Object Storage & MySQL Backend</h2>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b3122415",
+   "metadata": {},
+   "source": [
+    "**<span style='color:#005EB8'>Author:**</span> Ismail Syed</br>\n",
+    "**<span style='color:#005EB8'>Title:**</span> Senior Data Scientist EMEA</br>\n",
+    "**<span style='color:#005EB8'>Company:**</span> Oracle\n",
+    "\n",
+    "<span style='color:#005EB8'>**Notebook Description:**</span> In this Notebook we will look at how we can use the `oci-mlflow` library to save our models and model experiment data to our OCI MLFlow Tracking Server while using MLFlow Authentication and an OCI Object Storage & MySQL Backend for storing artifacts and experiemnts data.</br></br> <span style='color:#005EB8'>**NOTE:**</span> This sample code is intended for tutorial purposes only.</br>\n",
+    "\n",
+    "The data set used in this example is from http://archive.ics.uci.edu/ml/datasets/Wine+Quality\n",
+    "P. Cortez, A. Cerdeira, F. Almeida, T. Matos and J. Reis.\n",
+    "Modeling wine preferences by data mining from physicochemical properties. In Decision Support Systems, Elsevier, 47(4):547-553, 2009.\n",
+    "</br>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "047d77ec",
+   "metadata": {},
+   "source": [
+    "</br><h3 style='color:#005EB8'>Imports</h3>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cf5dd7fa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import logging\n",
+    "import os\n",
+    "import sys\n",
+    "import warnings\n",
+    "from urllib.parse import urlparse\n",
+    "\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "from sklearn.linear_model import ElasticNet\n",
+    "from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "import mlflow\n",
+    "import mlflow.sklearn\n",
+    "from mlflow.models import infer_signature\n",
+    "\n",
+    "logging.basicConfig(level=logging.WARN)\n",
+    "logger = logging.getLogger(__name__)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4b68a2fb",
+   "metadata": {},
+   "source": [
+    "</br><h3 style='color:#005EB8'>Establish Connection to Tracking Server</h3>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4609d257",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define Username and Password as ENV Variables\n",
+    "os.environ['MLFLOW_TRACKING_USERNAME'] = '<USERNAME_GOES_HERE>'\n",
+    "os.environ['MLFLOW_TRACKING_PASSWORD'] = '<PASSOWRD_GOES_HERE>'"
```

Comment:
Typo. Should read PASSWORD not PASSOWRD.

Reply:
Resolved.